manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -52,7 +66,7 @@ spec:
             readOnlyRootFilesystem: true
           {{- end }}
           env:
-          {{- if not .Values.runAsRoot }}
+          {{- if and (not .Values.runAsRoot) (not $unprivPort) }}
```

Comment:
Will this change the semantics on GKE1.22 plus ? I.e. will it result in Istiod binding to port 80 instead of 8080 ? And if yes, any backward incompat problems ( for examples logs showing a different port after upgrade, or authz policies no longer matching ) ?

Reply:
No. on 1.21-, they will be broken if they set a low port. So the chart defaults to 80->8080. When they enable this, it does not _automatically_ switch the 8080 to 80. They could, but its not OOTB.

Reply:
The alternative would be to have:
1.21-: root
1.22: sysctl

and never set targetPort by default. If users on 1.21 and older want nonRoot, they can explicitly configure the ports.

Pros:
* Matches gateway controller
* Better long term - users don't need to change port #s and get the long term desired behavior (bind to 80)

Cons:
* Users have degraded security posture on 1.21 by default